File: sort/selection_sort.c
#Selection Sort

#include<stdio.h>
int main(){

  int s,i,j,temp,a[20];

  printf("Enter total elements: ");  // total no of elements
  scanf("%d",&s);

  printf("Enter %d elements: ",s);  // the elements
  for(i=0;i<s;i++)
      scanf("%d",&a[i]);

  for(i=0;i<s;i++){
      for(j=i+1;j<s;j++){
           if(a[i]>a[j]){
               temp=a[i];    // Compare between 2 consecutive elements and swap them if they are not in ascending order
              a[i]=a[j];
              a[j]=temp;
           }
      }
  }

  printf("After sorting is: ");
  for(i=0;i<s;i++)            //  Print the sorted array
      printf(" %d",a[i]);

  return 0;
}

File: sort/selection_sort.py
# implementation of selection sort algorithm
# Author: RAHUL THAKUR
# Date: 02 December 2017

def selection_sort(list_param):

    size=len(list_param)
    outer_counter = 0
    while outer_counter < size:

        smallest=list_param[outer_counter]
        loc = outer_counter
        inner_counter = outer_counter + 1

        while inner_counter < size:

            if list_param [ inner_counter] < smallest:
                smallest=list_param[ inner_counter ]
                loc = inner_counter

            inner_counter = inner_counter + 1

        temp=list_param[ outer_counter ]
        list_param[ outer_counter ]=smallest
        list_param[loc]=temp
        outer_counter = outer_counter + 1

    return list_param

print("Enter the elements in the list")

List = list(map(int,input().split()))

print("After application of Selection Sort Algorithm")

print(selection_sort(List))
